source/extensions/common/wasm/wasm.cc

```diff
@@ -983,7 +983,42 @@ const ProtobufWkt::Struct* Context::getMetadataStructProto(MetadataType type,
   case MetadataType::ResponseRoute:
     return getRouteMetadataStructProto(encoder_callbacks_);
   case MetadataType::Node:
+    if (name == "_") {
```

Comment:
I think it would be better not to supply all the metadata at once, but just have some kind of switch based on keys that only returns individual fields.

Reply:
You can ask for individual bits of the metadata with the high level API by calling namedMetadataValue("_", "cluster") for example.